GRDB/Record/FetchableRecord+Decodable.swift

```diff
@@ -487,6 +491,46 @@ private struct _RowDecoder<R: FetchableRecord>: Decoder {
     }
 }
 
+private struct SingleValueRowDecoder<R: FetchableRecord>: SingleValueDecodingContainer {
+    var columnDecoder: ColumnDecoder<R>
+    var columnDecodingStrategy: DatabaseColumnDecodingStrategy
+    let codingPath: [any CodingKey] = []
+    
+    func decodeNil() -> Bool { columnDecoder.decodeNil() }
+    func decode(_ type: Bool.Type) throws -> Bool { try columnDecoder.decode(type) }
+    func decode(_ type: String.Type) throws -> String { try columnDecoder.decode(type) }
+    func decode(_ type: Double.Type) throws -> Double { try columnDecoder.decode(type) }
+    func decode(_ type: Float.Type) throws -> Float { try columnDecoder.decode(type) }
+    func decode(_ type: Int.Type) throws -> Int { try columnDecoder.decode(type) }
+    func decode(_ type: Int8.Type) throws -> Int8 { try columnDecoder.decode(type) }
+    func decode(_ type: Int16.Type) throws -> Int16 { try columnDecoder.decode(type) }
+    func decode(_ type: Int32.Type) throws -> Int32 { try columnDecoder.decode(type) }
+    func decode(_ type: Int64.Type) throws -> Int64 { try columnDecoder.decode(type) }
+    @available(macOS 15.0, iOS 18.0, watchOS 11.0, tvOS 18.0, visionOS 2.0, *)
+    func decode(_ type: Int128.Type) throws -> Int128 { try columnDecoder.decode(type) }
+    func decode(_ type: UInt.Type) throws -> UInt { try columnDecoder.decode(type) }
+    func decode(_ type: UInt8.Type) throws -> UInt8 { try columnDecoder.decode(type) }
+    func decode(_ type: UInt16.Type) throws -> UInt16 { try columnDecoder.decode(type) }
+    func decode(_ type: UInt32.Type) throws -> UInt32 { try columnDecoder.decode(type) }
+    func decode(_ type: UInt64.Type) throws -> UInt64 { try columnDecoder.decode(type) }
+    @available(macOS 15.0, iOS 18.0, watchOS 11.0, tvOS 18.0, visionOS 2.0, *)
+    func decode(_ type: UInt128.Type) throws -> UInt128 { try columnDecoder.decode(type) }
```

Comment:
The support for 128-bit ints requires a compiler version check:

```suggestion
#if compiler(>=6)
    @available(macOS 15.0, iOS 18.0, watchOS 11.0, tvOS 18.0, visionOS 2.0, *)
    func decode(_ type: Int128.Type) throws -> Int128 { try columnDecoder.decode(type) }
#endif
    func decode(_ type: UInt.Type) throws -> UInt { try columnDecoder.decode(type) }
    func decode(_ type: UInt8.Type) throws -> UInt8 { try columnDecoder.decode(type) }
    func decode(_ type: UInt16.Type) throws -> UInt16 { try columnDecoder.decode(type) }
    func decode(_ type: UInt32.Type) throws -> UInt32 { try columnDecoder.decode(type) }
    func decode(_ type: UInt64.Type) throws -> UInt64 { try columnDecoder.decode(type) }
#if compiler(>=6)
    @available(macOS 15.0, iOS 18.0, watchOS 11.0, tvOS 18.0, visionOS 2.0, *)
    func decode(_ type: UInt128.Type) throws -> UInt128 { try columnDecoder.decode(type) }
#endif
```

Reply:
Ah, should not have used beta Xcode for this.

Reply:
That'll be less work for GRDB7 :+1: